### PR TITLE
Add a new configuration to DoubleStartEndWith

### LINKED ISSUE #3893: How to handle ActiveSupport's starts_with? and ends_with? in Performance/DoubleStartEndWith
I recently discovered that [`ActiveSupport` adds in `starts_with?` and `ends_with?` as aliases to `start_with?` and `end_with?`](http://edgeguides.rubyonrails.org/active_support_core_extensions.html#starts-with-questionmark-and-ends-with-questionmark). I would like to add support for these methods, but I am not sure what the best way to accomplish this is.

One idea is to add a configuration to `Performance/DoubleStartEndWith` to enable checks for `starts_with?` and `ends_with?`.

Another idea is to add a Rails specific cop to handle this check separately from the Performance cop.

What is the preferred way to implement this check?

## RuboCop version

Any version that includes `Performance/DoubleStartEndWith`

### PR DESCRIPTION
**Pull request by rrosenblum**
This fixes #3893.

I noticed that this cop doesn't support auto-correct. After this PR is merged in, I can add support for auto-correct.

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -1292,6 +1292,11 @@ Lint/UnusedMethodArgument:
 
 ##################### Performance ############################
 
+Performance/DoubleStartEndWith:
+  # Used to check for `starts_with?` and `ends_with?`.
+  # These methods are defined by `ActiveSupport`.
+  IncludeDoubleStartsEndsWith: false
```
Comment (bbatsov) [inline config/default.yml]: I'd name this `IncludeActiveSupportAliases`.

File: lib/rubocop/cop/performance/double_start_end_with.rb [hunk 2]
```diff
@@ -33,6 +33,9 @@ def on_or(node)
           second_call_args = two_start_end_with_calls(node)
 
           return unless receiver && second_call_args.all?(&:pure?)
+          unless check_for_starts_with_and_ends_with?
+            return if [:starts_with?, :ends_with?].include?(method)
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/double_start_end_with.rb]: I'd move this array to a constant.

File: lib/rubocop/cop/performance/double_start_end_with.rb [hunk 3]
```diff
@@ -57,9 +60,15 @@ def add_offense_for_double_call(node, receiver, method, combined_args)
...
 
+        def check_for_starts_with_and_ends_with?
+          cop_config['IncludeDoubleStartsEndsWith']
+        end
+
         def_node_matcher :two_start_end_with_calls, <<-END
           (or
-            (send $_recv [{:start_with? :end_with?} $_method] $...)
+            (send $_recv
+                    [{:start_with? :starts_with? :end_with? :ends_with?} $_method]
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/double_start_end_with.rb]: Shouldn't we have two different matchers?

**Comment by rrosenblum** [comment]
I have updated the code based on the earlier comments. By using two different matchers, I wound up not needing a constant to store the ActiveSupport aliases. Let me know what you think of the changes.

**Comment by bbatsov** [comment]
The code looks good to me, but the build is failing.

**Comment by rrosenblum** [comment]
I have fixed the build and rebased from master.

**Pull request merged**